internal/provider/waypoint/resource_waypoint_application.go

```diff
@@ -272,9 +441,77 @@ func (r *ApplicationResource) Read(ctx context.Context, req resource.ReadRequest
 		data.ReadmeMarkdown = types.StringNull()
 	}
 
+	inputVars, err := clients.GetInputVariables(ctx, client, data.Name.ValueString(), loc)
+	if err != nil {
+		resp.Diagnostics.AddError(err.Error(), "Failed to fetch application's input variables.")
+		return
+	}
+
+	resp.Diagnostics.Append(readInputs(ctx, inputVars, data, varTypes)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Error(ctx, "error reading application input variables")
+		return
+	}
+
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
+// readInputs reads the inputVars for an app into the application resource model.
+func readInputs(
+	ctx context.Context,
+	inputVars []*waypoint_models.HashicorpCloudWaypointInputVariable,
+	plan *ApplicationResourceModel,
+	varTypes map[string]string,
+) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if inputVars != nil {
```

Comment:
we should probably omit this check and just always do this in order to detect drift.

Reply:
Fixed in https://github.com/hashicorp/terraform-provider-hcp/pull/833/commits/8dad09e6db677d8af2bf3df37dfd88e58698cc9c.